### ISSUE TITLE
Store whether Magnifier is enabled in the config

### ISSUE THREAD

**Issue by SaschaCowley** [nvaccess/nvda#19494]
### Is your feature request related to a problem? Please describe

Enabling NVDA's new magnifier feature does not cause this to be stored across restarts. For users who use the feature frequently, this is likely to be desired.

### Describe the solution you'd like

When toggling the magnifier, store whether it is enabled in NVDA's configuration. Add a checkbox to the Magnifier settings dialog to toggle the magnifier on and off.

### Describe alternatives you've considered

Implement something like Screen Curtain's temporary mode for the magnifier.

**Comment by CyrilleB79** [comment]
I strongly disagree with introducing another temporary mode. Having both temporary and persistent modes for the same feature is a bad UX. Either the feature should be persistent (as most of NVDA features), or temporary = per-session (as Speech mode).
Actually, I support clarifying Screen Curtain on this matter.

**Comment by Boumtchack** [comment]
Hi, I'm not sure, what you want is that, if the magnifier is on when NVDA is shut down, magnifier should start with nvda in next session to prevent users to use gestures?
Shouldn't we just had a checkbox with "lauch when NVDA start" in the magnifier settings?
It seems to me it would be better that way but it's your call

**Comment by seanbudd** [comment]
This would involve using the existing checkbox, but checking/unchecking whenever the magnifier is enabled or disabled. It's expected that most users would either be leaving it on or off, so either way it will be persisted in settings as expected.

**Comment by CyrilleB79** [comment]
@seanbudd you wrote:
> This would involve using the existing checkbox, but checking/unchecking whenever the magnifier is enabled or disabled.

Which "existing checkbox"?

This seems unconsistent with what Sascha wrote in the initial description mentioning a new checkbox, not an existing one:
> Add a checkbox to the Magnifier settings dialog to toggle the magnifier on and off.

**Comment by seanbudd** [comment]
@CyrilleB79 - I meant the checkbox proposed, rather than @Boumtchack's suggestion
